cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/resource/AbstractReadyOperations.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.controller.cluster.operations.resource;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesResourceList;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.client.internal.readiness.Readiness;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractReadyOperations<C, T extends HasMetadata, L extends KubernetesResourceList/*<T>*/, D, R extends Resource<T, D>, P>
+        extends AbstractOperations<C, T, L, D, R, P> {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    /**
+     * Constructor.
+     *
+     * @param vertx        The vertx instance.
+     * @param client       The kubernetes client.
+     * @param resourceKind The mind of Kubernetes resource (used for logging).
+     */
+    public AbstractReadyOperations(Vertx vertx, C client, String resourceKind) {
+        super(vertx, client, resourceKind);
+    }
+
+    /**
+     * Returns a future that completes when the resource identified by the given {@code namespace} and {@code name}
+     * is ready.
+     *
+     * @param namespace The namespace.
+     * @param name The resource name.
+     * @param pollIntervalMs The poll interval in milliseconds.
+     * @param timeoutMs The timeout, in milliseconds.
+     */
+    public Future<Void> readiness(String namespace, String name, long pollIntervalMs, long timeoutMs) {
+        Future<Void> fut = Future.future();
+        log.info("Waiting for {} resource {} in namespace {} to get ready", resourceKind, name, namespace);
+        long deadline = System.currentTimeMillis() + timeoutMs;
+        Handler<Long> handler = new Handler<Long>() {
+            @Override
+            public void handle(Long timerId) {
+                vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
+                    future -> {
+                        try {
+                            if (isReady(namespace, name))   {
+                                future.complete();
+                            } else {
+                                if (log.isTraceEnabled()) {
```

Comment:
hm, I've never seen the using of method `.isTraceEnabled()` before for cases when the string has placeholders (because slf4j should handle this), but after reading the documentation it seems that using  `.isTraceEnabled()` is necessary to increase performance if we have expensive parameters. 
P.S. (just thinking)

Reply:
Yeah, this is a good point. Partly it comes from the days where you couldn't pass a `"template {} string"` to loggers, and had to concatenate your message yourself in the call to `trace()` (or whatever log level). That meant that you spent time allocating and concatenating strings, only for the logging method to be a no-op. There's less need for this these days because passing a template string means that concat doesn't happen if the trace logging isn't enabled. There's still the cost of array allocation for the var args, but in reality the jit inlining should end up optimising the call away anyway, eventually.